### ISSUE TITLE
Experiment with Path2D for improved performance

### ISSUE THREAD

**Issue by kangax** [fabricjs/fabric.js#1263]
http://blogs.adobe.com/webplatform/2014/04/01/new-canvas-features/
##

**Comment by melchiar** [comment]
I just did some basic testing and Path2D is indeed significantly faster than the current method when drawing a complex path. 
![path-rendering-current](https://user-images.githubusercontent.com/9057412/78406394-68722f00-75b7-11ea-92fe-90c68b2f33ce.gif)
![path-rendering-path2d](https://user-images.githubusercontent.com/9057412/78406401-6c05b600-75b7-11ea-90d7-9b500da4352f.gif)

**Comment by asturur** [comment]
yes it is without doubt quicker and saves memory and time from our parsing and caching of arcs.

If someone wants to make a pr to have both of them i m happy to review it.

**Comment by melchiar** [comment]
one difficulty is how to accurately detect full browser support for Path2D. Apparently Edge only has partial support, and doesn't allow parsing of svg path strings.

https://stackoverflow.com/questions/53787259/detect-when-path2d-with-svg-path-string-is-supported

**Comment by asturur** [comment]
Well there are 2 things at stake here, speed and code semplification.
Now in my mind there was always the idea to simply SVG paths, since we can't draw the A segments and we need to parse them in multiple C segments.

Also SVG can have the lowercase and uppercase commands.

If you want to fiddle with path commands, having everything uppercase and not having As is easier.

So we could anyway support path2d after semplifying the paths and manually adding them to the Path2D object. That would give us speed anyway but more code more than less of it.
